insights/parsers/partitions.py

```diff
@@ -0,0 +1,60 @@
+"""
+Partitions - file ``/proc/partitions``
+======================================
+
+This parser reads the ``/proc/partitions`` file, which contains
+partition block allocation information.
+
+"""
+from insights import Parser, parser
+from insights.parsers import SkipException, parse_delimited_table
+from insights.specs import Specs
+
+
+@parser(Specs.partitions)
+class Partitions(Parser):
+    """A parser class for parsing the ``/proc/partitions`` file.
+
+    Sample input::
+
+        major minor  #blocks  name
+
+           3     0   19531250 hda
+           3     1     104391 hda1
+           3     2   19422585 hda2
+         253     0   22708224 dm-0
+         253     1     524288 dm-1
+
+    Examples:
+       >>> type(partitions)
+       <class 'insights.parsers.partitions.Partitions'>
+       >>> partitions.data[0].get('name')
+       'hda'
+       >>> partitions.data[0].get('major')
+       '3'
+       >>> partitions.get_blocks('dm-0')
+       '22708224'
+
+    Raises:
+        SkipException: When input is empty.
+    """
+    def parse_content(self, content):
+        if not content:
+            raise SkipException('Empty content')
+
+        self.data = parse_delimited_table(content,
```

Comment:
@akshay196 the `data` attribute is an old name that would be better not to use because is too generic.  What about using the name `partitions` or something more descriptive of the data that is being provided by the parser?

Also would it be better for the attribute to be a `dict` of partitions so that it could be indexed by name?  You could present it as both a list and a dict if that is how you anticipate it being accessed.  For instance you could create an attribute of partition names and implement `__getitem__` to index by partition name and `__contains__` to allow checking using the `in` operator.

Reply:
Done. Updated code now. 
@bfahr Thanks for suggestion.